docs/alerting-recording.md

```diff
@@ -0,0 +1,63 @@
+# Alerting & Recording Rules
+
+### Alerting Rules
+
+Alerting rules are used to trigger alerts based on the violation of any condition(s). These alerts are fired to external services like slack, mails, etc by the [alert manager](https://prometheus.io/docs/alerting/latest/alertmanager/). Alerting rules are written in a YAML file and paths of these files are mentioned in the Prometheus configuration respectively. It is important to note that the evaluation of these conditional rules are done at the Prometheus side.
+
+```
+groups:
+  - name: <alert-group-name>
+    rules:
+    - alert: <alert-name>
+      expr: <promql_expression>
+      for: <time-interval for how long this to happen to happen to fire an alert>
+      labels:
+        <key>:<value>
+      annotations:
+        summary: <text>
+        description: <description on alert>
+```
+
+Load the above defined alert rule file to your prometheus configuration for evaluation.
+
+```
+rule_files:
+  - "<alert-rules-file>"
+```
+
+Configure the alert manager in prometheus configuration.
+
+```
+alerting:
+  alertmanagers:
+  - static_configs:
+    - targets:
+      - localhost:9093
+```
+
+More details on alerting rules can be found [here](https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/).
+
+
+### Recording Rules
+
+The recording rules are for pre-calculating frequently used or computationally expensive queries. The results of those rules are saved into their own time series. The newly formed time-series will follow the name of the record. In promscale the rules are evaluated in prometheus i.e define recording rules in prometheus configuration.
```

Comment:
these rules are fired in prometheus but will they fetch data from promscale if needed or does the data retention in prometheus have to be long enough to contain all the data. we should clarify.

Also we should clarify that the new series is written into promscale.

Reply:
If `read_recent: true` is configured under `remote-read` configuration of prometheus. Every read request is forwarded to remote-read. The alerting & recording rules evaluation is performed by querying the data from remote-read. 

If disabled the data from promethues will be used. In both alerting & recording rules the evaluation happens for every `15s` or can be configured and the new series are written to promscale. The data retention policies configured should be higher than the time interval specified in alerting & recording rules (only if `read_recent` is disabled.)

Reply:
Then let's mention users use `read_recent: true` as recommended over here and in the `note` section.